### PR TITLE
Add d3drm

### PR DESCRIPTION
**Pull request by Zemogiter**
Verb is working and ready for review.

### CODE REVIEW

File: Engines/Wine/Verbs/d3drm/script.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        .checksum("a97c820915dc20929e84b49646ec275760012a42")
+        .name("directx_feb2010_redist.exe")
+        .get();
+    this.wizard().wait(tr("Please wait while {0} is installed ...", "d3drm"));
+    new CabExtract()
+        .archive(setupFile)
+        .to(this.prefixDirectory() + "/drive_c/d3drm/")
+        .extract(["-L", "-F", "dxnt.cab"]);
+    new CabExtract()
+        .archive(this.prefixDirectory() + "/drive_c/devenum/dxnt.cab")
```
Comment (madoar) [inline Engines/Wine/Verbs/d3drm/script.js]: This looks strange to me, where comes the `/drive_c/devenum/dxnt.cab` file from?

Comment (Zemogiter) [reply Engines/Wine/Verbs/d3drm/script.js]: From the previous `CabExtract`. The verb have to extract `dxnt.cab` from the .exe file it downloads. Then it extract the desired .dll file from that .cab archive.

Comment (madoar) [reply Engines/Wine/Verbs/d3drm/script.js]: Yes but take a look at the folder name.
As far as I understand your code you first extract the `dxnt.cab` file to `/drive_c/d3drm/dxnt.cab`.
Then you try to extract the `d3drm.dll` from `/drive_c/devenum/dxnt.cab`.
The folder names are different.

Comment (Zemogiter) [reply Engines/Wine/Verbs/d3drm/script.js]: You are correct.

File: Engines/Wine/Verbs/d3drm/script.js [hunk 2]
```diff
@@ -32,3 +32,20 @@ Wine.prototype.d3drm = function () {
 
     return this;
 };
+
+/**
+ * Verb to install all the necessary things to run winevulkan (even inside wine mainline or newest wine-staging)
```
Comment (plata) [inline Engines/Wine/Verbs/d3drm/script.js]: copy paste error

File: Engines/Wine/Verbs/d3drm/script.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  
+    this.overrideDLL()
+        .set("native", ["d3drm"])
+        .do();
+
+    return this;
+};
+
+/**
+ * Verb to install all the necessary things to run d3drm (even inside wine mainline or newest wine-staging)
```
Comment (plata) [inline Engines/Wine/Verbs/d3drm/script.js]:
```suggestion
 * Verb to install d3drm
```

**Comment by Zemogiter** [comment]
There is an issue though. Installing any verb using the "Verbs" tab results in this error:
```
javax.script.ScriptException: ReferenceError: "Verb" is not defined in <eval> at line number 1
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.throwAsScriptException(NashornScriptEngine.java:477)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:461)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:413)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:409)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:162)
	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornEngine.evalAndReturn(NashornEngine.java:59)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornInteractiveSession.eval(NashornInteractiveSession.java:34)
	at org.phoenicis.scripts/org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$createInteractiveSession$1(BackgroundScriptInterpreter.java:43)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: <eval>:1 ReferenceError: "Verb" is not defined
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.error(ECMAErrors.java:57)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.referenceError(ECMAErrors.java:319)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.referenceError(ECMAErrors.java:291)
	at jdk.scripting.nashorn/jdk.nashorn.internal.objects.Global.__noSuchProperty__(Global.java:1616)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$280$\^eval\_/0x00000008408e0840.:program(<eval>:1)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:655)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:513)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:527)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:456)
	... 10 more
```

**Comment by plata** [comment]
Only for this Verb or do you have the issue for all Verbs? If it works for others, compare your implementation. Otherwise, ensure your software is up-to-date.

**Comment by Zemogiter** [comment]
Updated to the latest version, still getting the same error.

**Comment by plata** [comment]
Only this Verb or also others?

**Comment by plata** [comment]
Strange.

**Comment by Zemogiter** [comment]
Any way to fix it?

**Comment by plata** [comment]
I have no idea currently. The error would make sense if your software is not up-to-date but if that's the case...

**Comment by Zemogiter** [comment]
I've just retested `amstream`,`d3drm` and `devenum` alongside VK9 and corefonts after I've found out there were some changes to the POL5 repo. Everything now works and can be merged.

**Pull request merged**